### PR TITLE
Fix documentation front matter and merge it with index.rst

### LINKED ISSUE #902: Documentation front matter is outdated
The documentation front matter in `front.rst` needs to be updated: it still refers to the old Enthought address (515 Congress).

### PR DESCRIPTION
**Pull request by ievacerny**
Closes #902 

The PR follows TraitsUI approach by merging `front.rst` into `index.rst` (TraitsUI PR https://github.com/enthought/traitsui/pull/778)

**Checklist**
- ~[ ] Tests~
- ~[ ] Update API reference (`docs/source/traits_api_reference`)~
- [x] Update User manual (`docs/source/traits_user_manual`)
- ~[ ] Update type annotation hints in `traits-stubs`~

### CODE REVIEW

File: docs/source/traits_user_manual/index.rst [hunk 1]
```diff
@@ -1,10 +1,17 @@
+====================
 Traits 6 User Manual
 ====================
 
+:Authors: David C. Morrill, Janet M. Swisher, and Enthought developers
+:Version: Document Version 4
```
Comment (mdickinson) [inline docs/source/traits_user_manual/index.rst:6]: I'm not sure whether this line is meaningful or useful. Should we just drop it?

File: docs/source/traits_user_manual/index.rst [hunk 2]
```diff
@@ -21,3 +28,42 @@ Indices and tables
 
 * :ref:`genindex`
 * :ref:`search`
+
+
+License
```
Comment (mdickinson) [inline docs/source/traits_user_manual/index.rst:33]: Would it make sense to move the License stuff above the indices? I think I expect to find the index as the last item on the page, but others may have different expectations.

File: docs/source/traits_user_manual/index.rst (hunk 1 above)
Comment (ievacerny) [reply docs/source/traits_user_manual/index.rst:6]: It hasn't been updated for 12 years so I it's probably not important

File: docs/source/traits_user_manual/index.rst (hunk 2 above)
Comment (corranwebster) [reply docs/source/traits_user_manual/index.rst:33]: I would have it at the bottom after the index reference, I think.  Certainly that's what I did for TraitsUI.

Comment (mdickinson) [reply docs/source/traits_user_manual/index.rst:33]: Okay, SGTM

File: docs/source/traits_user_manual/index.rst (hunk 1 above)
Comment (mdickinson) [reply docs/source/traits_user_manual/index.rst:6]: Yep, let's remove this.

**Pull request merged**